pkg/clusteragent/admission/mutate/autoinstrumentation/config.go

```diff
@@ -38,8 +38,12 @@ type Config struct {
 	containerRegistry string
 
 	// precomputed mutators for the security and profiling products
-	securityClientLibraryPodMutators  []podMutator
-	profilingClientLibraryPodMutators []podMutator
+	securityClientLibraryMutator  containerMutator
+	profilingClientLibraryMutator containerMutator
+
+	// precomputed containerPredicate to use with filteredContainerMutator
+	// and other mutators
+	containerFilter containerPredicate
```

Comment:
This isn't super descriptive. What kind of filter? What are we filtering and why? When should I use this?